### DISCUSSION TITLE
improve performance db2view

### DISCUSSION THREAD

**Discussion by merinet** [cea-sec/ivre#1495] [Q&A] [answered]
Hi,
Is there any way to improve performance for indexing process? It's taking weeks to finish indexing of some millions of records (running ivre db2view ). MongoDB seems be fine, but IVRE process is using just 1 CPU. Is there any way to improve this task?

Regards,

**Comment by p-l-** [comment, marked as answer]
That's a good question, and short answer is: no, apart from providing better code.

There is clearly room for improvement and as you mentioned the code is not parallel at all. I'm not sure it would be useful to parallelize in Python, because of the GIL. You may want to open an issue, rather that a question, to see if someone is interested in working on it more.
